### ISSUE TITLE
About the relevant issues when I run the trsnssc model with the PyTorch version of the code

### ISSUE THREAD

**Issue by biubiubiu1104** [cocoon2wong/SocialCircle#7]
Hi,
when I run the transsc model of the PyTorch version, an error "Boolean value of Tensor with more than one value is ambiguous" occurs. Here is my solution.

Change the code at line 61 of the __loss.py file from `pred = pred[..., indices, :]` to `pred = pred[..., len(indices), :]`  # The forward method of the avgKey class. Ensure that scalar values are compared instead of the entire Tensor.

At this time, another error "only integer tensors of a single element can be converted to an index" occurs, and the error occurs in the forward method of the ADE class in the __layers.py file. Specifically, it is at `points` (it may be a Tensor with multiple values rather than a single integer scalar, so the above error is raised when performing slicing operations).

Change

if torch.all(points > 0):
    pred = pred[..., :points, :]
    label = label[..., :points, :]
    obs = label[..., :points, :]

to

if isinstance(points, int) and points > 0:  # Ensure that points is an integer
    pred = pred[..., :points, :]
    label = label[..., :points, :]
    obs = label[..., :points, :]
elif isinstance(points, torch.Tensor) and points.numel() == 1:  # Ensure that points is a Tensor with a single value
    points = points.item()
    if points > 0:
        pred = pred[..., :points, :]
        label = label[..., :points, :]
        obs = label[..., :points, :]

Thanks!

**Comment by cocoon2wong** [comment]
Hi @biubiubiu1104 !

Thank you for your questions!
I have tried to reproduce this problem using the command
```bash
python main.py --sc ../Project-Monandaeg-SC/transsczara1
```
 with our provided weights `transsczara1` (https://github.com/cocoon2wong/Project-Monandaeg/tree/SocialCircle), while it works fine.

But this error raised when I tried to test this single model with
```bash
python main.py --load ../Project-Monandaeg-SC/transsczara1
```

I have checked the codes when computing this loss (metric), and I noticed that it is caused by the `ADE.forward` call that has received parameters with a wrong order. A temporary fix could be to replace lines 65-66 in the `qpid/silverballers/__loss.py` with
```python
        return super().forward(outputs=[pred],
                               labels=[labels_key],
                               inputs=inputs,
                               mask=mask, training=training,
                               *args, **kwargs)
```

This error is caused by the recent `qpid` updates in this repo. I have created a temporary branch to fix this error (in the torch branch). You can update your codes via
```bash
git pull
git submodule update --init --recursive
``` 

I hope these could help!
Conghao Wong

**Comment by biubiubiu1104** [comment]
Hi @cocoon2wong ，
Thank you for your response. I did try the method you suggested yesterday, but I found that the issue was not completely resolved. There is no ADE-related information in TensorBoard. I will try your approach when I have some free time.
Thanks！

**Comment by biubiubiu1104** [comment]
Hi @cocoon2wong ,
Thanks for your reply. Yesterday, I used my method and found that the problem was not completely solved. There was no ADE related information in Tensorboard, so I will use your method in my spare time.
Thanks!
